### ISSUE TITLE
User assertions do not work anymore

### ISSUE THREAD

**Issue by kotnik** [keybase/client#1630]
After upgrade to Keybase 1.0.7, user assertions do not work anymore:

```
$ keybase pgp verify --infile .public_keys/user.key --detached .public_keys/user.key.asc --signed-by "someuser@github || someother@github"
▶ ERROR assertion parse error: Unexpected 'OR' operator
```

There is really no documentation on this, and in the source code I see:

```go
func NewLexer(s string) *Lexer {
	// We're allowing '||' or ',' for disjunction
	// We're allowing '&&' or '+' for conjunction
```

So I guess this has been broken.

**Comment by maxtaco** [comment]
I agree, we'll fix it. Thanks for the bug report. cc: @patrickxb

**Comment by patrickxb** [comment]
Signed by uses Identify to make user assertions, and Identify only allows ANDs:

    func (e *Identify) loadExpr(assertion string) error {
	    // Parse assertion but don't allow OR operators, only AND operators
	    expr, err := libkb.AssertionParseAndOnly(assertion)

**Comment by maxtaco** [comment]
Right, but we should load the user by fingerprint (not by assertion) and then match the assertion to the full ProofSet.
